### PR TITLE
fix #136: [CA] Add etymology

### LINKED ISSUE #136: [CA] Add etymology
Add etymologies for the catalan locale.

Models:
- [-etimologia-](https://ca.wiktionary.org/wiki/Plantilla:-etimologia-)

### PR DESCRIPTION
**Pull request by lasconic**
- [X] Fixes #136
- [X] Tests added/updated.
- [X] `./check.sh` run and eventual issues fixed.
- [X] Updated CONTRIBUTORS.md, if not already done in the past.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Is it ready for a review?

**Comment by lasconic** [comment]
Sourcery is not happy. Let me make him happy and it should be ok.

**Comment by lasconic** [comment]
Ready for review now

**Review by BoboTiG — COMMENTED** [review]
Thanks. I'll have time in the weekend for a review :)

File: scripts/get.py [hunk 1]
```diff
@@ -263,9 +267,28 @@ def find_all_sections(code: str, locale: str) -> List[wtp.Section]:
...
+    if not isinstance(etyl_l_sections, list):
+        etyl_l_sections = [etyl_l_sections]  # type: ignore
+    for etyl_l_section in etyl_l_sections:
+        for line in leading_lines:
+            if line.startswith(etyl_l_section):
+                etyl_data = line
+                etyl_data_section = etyl_l_section
+                break
+    if etyl_data:
+        sections.append(wtp.Section("=== " + etyl_data_section + " ===\n" + etyl_data))
```
Comment (BoboTiG) [inline scripts/get.py:289]:
```suggestion
        sections.append(wtp.Section(f"=== {etyl_data_section} ===\n{etyl_data}"))
```

File: scripts/lang/ca/__init__.py [hunk 2]
```diff
@@ -35,7 +36,10 @@
     "Sufix",
     "Símbol",
     "Verb",
-)
+    *etyl_section,
+]
+
+sections = tuple(l_sections)
```
Comment (BoboTiG) [inline scripts/lang/ca/__init__.py:42]:
```suggestion
)
```

File: scripts/lang/ca/__init__.py [hunk 3]
```diff
@@ -16,7 +16,8 @@
 
 # Markers for sections that contain interesting text to analyse.
 head_sections = ("{{-ca-}}", "{{-mul-}}")
-sections = (
+etyl_section = ["{{-etimologia-", "{{-etim-", "{{etim-lang"]
+l_sections = [
```
Comment (BoboTiG) [inline scripts/lang/ca/__init__.py:20]:
```suggestion
sections = (
```

File: scripts/lang/ca/__init__.py [hunk 4]
```diff
@@ -89,6 +93,10 @@
     "q": "term(parts[-1])",
     # {{etim-s|ca|XIV}}
     "etim-s": "'segle ' + parts[2]",
+    # {{-etimologia-|ca|es}}
+    "-etimologia-": "strong('Etimologia:')",
```
Comment (BoboTiG) [inline scripts/lang/ca/__init__.py:97]: Hm this is nt good. Either there is something either nothing. We do not need to have "Etimologia:" printed. It is already handled homogeneously for all locales. Also, it does not seem to be the good handling for that template?

**Review by BoboTiG — COMMENTED** [review]
You are missing the most important tests in `test_ca.py::test_parse_word()`. See how etymology has been added to such tests here: https://github.com/BoboTiG/ebook-reader-dict/blob/e1e632d23a5372a63f48be3907088d781a1f5a47/tests/test_fr.py#L308

File: scripts/lang/ca/__init__.py (hunk 4 above)
Comment (lasconic) [reply scripts/lang/ca/__init__.py:97]: It is the good handling... at least on wiktionary. See https://ca.wiktionary.org/wiki/Plantilla:-etimologia- and [Dijous](https://ca.wiktionary.org/w/index.php?title=dijous&action=raw) // [Dijous](https://ca.wiktionary.org/wiki/dijous) for example. 

But, sure, we don't really need it here. So let's remove it entirely.

Comment (BoboTiG) [reply scripts/lang/ca/__init__.py:97]: Oh yes, I misread the documentation. But indeed they are not needed :)

**Comment by lasconic** [comment]
I addressed the remarks and added etymology in the tests.

File: tests/test_ca.py [hunk 5]
```diff
@@ -90,6 +102,7 @@
             "català",
             "",
             "m",
+            "D’origen incert, paral·lel al de <i>Catalunya</i>, possiblement metàtesi del llatí <i>lacetani</i> («lacetans»).",
```
Comment (BoboTiG) [inline tests/test_ca.py:105]:
```suggestion
            "D’origen incert, paral·lel al de <i>Catalunya</i>, possiblement metàtesi del llatí <i>lacetani</i> («lacetans»).",  # noqa
```

**Comment by BoboTiG** [comment]
Merci beaucoup :)

**Pull request merged**